test/widgets/button_test.dart

```diff
@@ -114,6 +114,39 @@ void main() {
       check(renderObject).size.equals(Size.square(40));
     });
 
+    group('AnimatedScaleOnPrimaryPointerDown', () {
+      void checkScale(WidgetTester tester, Finder finder, double expectedScale) {
+        final scale = tester.widget<AnimatedScale>(finder).scale;
+        check(scale).equals(expectedScale);
+      }
+
+      testWidgets('Animation happen instantly when tap down', (tester) async {
+        addTearDown(testBinding.reset);
+
+        await tester.pumpWidget(TestZulipApp(
+          child: AnimatedScaleOnPrimaryPointerDown(
+            scaleEnd: 0.95,
+            duration: Duration(milliseconds: 100),
+            child: UnconstrainedBox(
+              child: ZulipIconButton(
+                icon: ZulipIcons.follow,
+                onPressed: () {})))));
+        await tester.pump();
+
+        final animatedScaleFinder = find.byType(AnimatedScale);
+
+        // Now that the widget is being held down, its scale should be at the target scaleEnd i.e 0.95.
+        final gesture = await tester.startGesture(tester.getCenter(find.byType(ZulipIconButton)));
+        await tester.pumpAndSettle(const Duration(milliseconds: 50));
```

Comment:
Why `pumpAndSettle`? From its dartdoc:

```dart
  /// Repeatedly calls [pump] with the given `duration` until there are no
  /// longer any frames scheduled. This will call [pump] at least once, even if
  /// no frames are scheduled when the function is called, to flush any pending
  /// microtasks which may themselves schedule a frame.
```

The `AnimatedScale` should get set to 0.95 right away, right, and it handles the animation internally. How about:

```suggestion
        await tester.pump();
```